src/js/components/AccordionPanel/AccordionPanel.js

```diff
@@ -14,133 +14,125 @@ import { withForwardRef } from '../hocs';
-    this.setState({ hover: undefined });
-    if (onMouseOut) {
-      onMouseOut(args);
-    }
-  };
-
-  onFocus = (...args) => {
-    const {
-      onFocus,
-      theme: { dark },
-    } = this.props;
-    this.setState({ hover: dark ? 'light-4' : 'dark-3' });
-    if (onFocus) {
-      onFocus(args);
-    }
-  };
-
-  onBlur = (...args) => {
-    const { onBlur } = this.props;
-    this.setState({ hover: undefined });
-    if (onBlur) {
-      onBlur(args);
-    }
-  };
-
-  render() {
-    const {
-      children,
-      header,
-      label,
-      theme,
-      onMouseOut,
-      onMouseOver,
-      onFocus,
-      onBlur,
-      ...rest
-    } = this.props;
-    const { hover } = this.state;
-
-    const iconColor = normalizeColor(
-      theme.accordion.icons.color || 'control',
-      theme,
-    );
-
-    return (
-      <AccordionContext.Consumer>
-        {panelContext => {
-          const { active, animate, onPanelChange } = panelContext;
-          const AccordionIcon = active
-            ? theme.accordion.icons.collapse
-            : theme.accordion.icons.expand;
-
-          return (
-            <Box flex={false}>
-              <Button
-                role="tab"
-                aria-selected={active}
-                aria-expanded={active}
-                onClick={onPanelChange}
-                onMouseOver={this.onMouseOver}
-                onMouseOut={this.onMouseOut}
-                onFocus={this.onFocus}
-                onBlur={this.onBlur}
-              >
-                {header || (
-                  <Box
-                    align="center"
-                    direction="row"
-                    justify="between"
-                    {...rest}
-                  >
-                    {typeof label === 'string' ? (
-                      <Box pad={{ horizontal: 'xsmall' }}>
-                        <Heading
-                          level={
-                            (theme.accordion.heading &&
-                              theme.accordion.heading.level) ||
-                            4
-                          }
-                          color={hover}
-                        >
-                          {label}
-                        </Heading>
-                      </Box>
-                    ) : (
-                      label
-                    )}
-                    {AccordionIcon && (
-                      <Box pad={{ horizontal: 'small' }}>
-                        <AccordionIcon color={iconColor} />
-                      </Box>
-                    )}
-                  </Box>
-                )}
-              </Button>
-              <Box border={theme.accordion.border}>
-                {animate ? (
-                  <Collapsible open={active}>{children}</Collapsible>
-                ) : (
-                  active && children
-                )}
-              </Box>
+const onHandleMouseOver = ({ onMouseOver, theme, setHover }) => (...args) => {
```

Comment:
any particular reason to have those functions outside of the component? the code will be a bit simpler without needing to provide those parameters in all the functions. not sure the performance gains are even notable to justify having them outside. thoughts?

Reply:
In my opinion the better solution always is the component just render the element, all the others logics should be outside the component.

Reply:
I agree with @oorestisime, l think the code could be more readable and clean having the functions inside the component.